only keep chr number when running samtools depth

tdelhomme opened the pull request:
Add a line to only keep chromosome number when reading output of samtools depth (chr17 -> 17).
If not, no consistency with the `regionDatabaseRef` and `offtargetdepth` and `ontargetdepth` are setting to `NA`

slzhao commented:
Hi Tiffany,

Thanks for your changes. I am thinking about "nod" option. For the depth
off/on target regions, there is on target depth/off target depth in the
report. Adding "nod" means we will only analyze on target depth so that we
can make bam QC faster? The on target depth in the report will be the same
and the off target depth in the report will be 0. Is it correct? And I also
found that you removed the "-a" option in samtools depth. I think we need
"-a" if we want to include the regions with 0 depth. What do you think?

Thanks again!

Best,
Shilin

2016-05-09 10:17 GMT-05:00 Tiffany Delhomme notifications@github.com:

> Add a line to only keep chromosome number when reading output of samtools
> depth (chr17 -> 17).
> If not, no consistency with the regionDatabaseRef and offtargetdepth and
> 
> ## ontargetdepth are setting to NA
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/slzhao/QC3/pull/7
> Commit Summary
> - only keep chr number when run samtools depth
> 
> File Changes
> - _M_ source/bamSummary.pm
>   https://github.com/slzhao/QC3/pull/7/files#diff-0 (1)
> 
> Patch Links:
> - https://github.com/slzhao/QC3/pull/7.patch
> - https://github.com/slzhao/QC3/pull/7.diff
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> https://github.com/slzhao/QC3/pull/7

tdelhomme commented:
Yes you are right:
- removing `-a` was a mistake (I removed it for testing) sorry I will correct that and propose a pull request.
- yes the idea of `-nod` is to save time by giving a bed file to samtools for only compute depth in bed regions (in some analyses, the off-target region is not the main interest but saving time could be much more important, and also for target sequencing for ex the off-target regions is all the remaining genome and this computation will take a lot of time, particularly because we added the `-a` option). Note that this option needs to have a bed file in input.
- yes the on target depth does not change with `-nod` and the off-target will be `NA` in the report.

I am trying to add an option `-no_batch` to do not compute the figures for the `batch effect` part. The idea is that depending on the bam building all informations concerning batch could be missing and this will create a bug when running.
I will let you know!

Thanks Shilin,

BW,
Tiffany
